File: backend/storage.py
import time

_DB = {}

def save_project(user_id, project):
    if not user_id:
        return
    record = {
        **project,
        "created_at": time.time(),
    }
    _DB.setdefault(user_id, []).append(record)

def get_projects(user_id):
    if not user_id:
        return []
    items = _DB.get(user_id, [])
    # Нормализация: image → images для квартиры
    for item in items:
        if "images" not in item and "image" in item:
            item["images"] = [item["image"]]
    return sorted(items, key=lambda x: x.get("created_at", 0), reverse=True)

File: backend/storage.js
const db = {};

function saveProject(userId, project) {
  if (!userId) return;
  const record = {
    ...project,
    created_at: Date.now() / 1000,
  };
  if (!db[userId]) db[userId] = [];
  db[userId].push(record);
}

function getProjects(userId) {
  if (!userId) return [];
  const items = (db[userId] || []).map((item) => ({ ...item }));
  for (const item of items) {
    if (!item.images && item.image) item.images = [item.image];
  }
  return items.sort((a, b) => (b.created_at || 0) - (a.created_at || 0));
}

function clearProjects(userId) {
  if (!userId) return;
  db[userId] = [];
}

module.exports = { saveProject, getProjects, clearProjects };
